src/main/java/org/manifold/compiler/front/ExpressionVisitor.java

```diff
@@ -2,9 +2,20 @@
 
 public interface ExpressionVisitor {
 
-  void visit(FunctionInvocationExpression functionInvocationExpression);
-  void visit(LiteralExpression literalExpression);
-  void visit(VariableAssignmentExpression variableAssignmentExpression);
-  void visit(VariableReferenceExpression variableReferenceExpression);
+  void visit(FunctionInvocationExpression functionInvocationExpression)
+    throws Exception;
+  void visit(LiteralExpression literalExpression) throws Exception;
+  void visit(VariableAssignmentExpression variableAssignmentExpression) 
+      throws Exception;
+  void visit(VariableReferenceExpression variableReferenceExpression) 
+      throws Exception;
+  void visit(PrimitivePortDefinitionExpression
+      primitivePortDefinitionExpression) throws Exception;
+  void visit(PrimitiveNodeDefinitionExpression
+      primitiveNodeDefinitionExpression) throws Exception;
+  void visit(TupleTypeValueExpression tupleTypeValueExpression) 
+      throws Exception;
+  void visit(FunctionTypeValueExpression functionTypeValueExpression) 
+      throws Exception;
```

Comment:
I think we should have a discussion about the merits of defining primitive ports and nodes in the frontend vs a separate config file. My only concern is that end users should basically *never* touch this part of the language and might be confused as to why its there.